Backend/Remora.Discord.Caching/Services/CacheSettings.cs

```diff
@@ -23,56 +23,67 @@
 using System;
 using System.Collections.Generic;
 using JetBrains.Annotations;
-using Microsoft.Extensions.Caching.Distributed;
-using Microsoft.Extensions.Caching.Memory;
 
 namespace Remora.Discord.Caching.Services;
 
 /// <summary>
 /// Holds various settings for individual cache objects.
 /// </summary>
+/// <remarks>
+/// Cache services should use <see cref="ImmutableCacheSettings"/> instead.
```

Comment:
Is this type deprecated due to this? If so, we should probably just delete it outright since the PR contains API breaking changes anyway.

Reply:
No, since the ImmutableCacheSettings derives from CacheSettings. The user configures CacheSettings via DI, but cache services themselves consuem the Immutable one.